packages/astro/test/custom-fetch-error-pages.test.js

```diff
@@ -0,0 +1,92 @@
+import assert from 'node:assert/strict';
+import { before, describe, it, beforeEach } from 'node:test';
+import * as cheerio from 'cheerio';
+import testAdapter from './test-adapter.js';
+import { loadFixture } from './test-utils.js';
+
+describe('Custom Fetch for Error Pages', () => {
+  /** @type {import('./test-utils.js').Fixture} */
+  let fixture;
+  
+  before(async () => {
+    fixture = await loadFixture({
+      root: './fixtures/custom-fetch-error-pages/',
+      output: 'server',
+      adapter: testAdapter(),
+      build: { inlineStylesheets: 'never' },
+    });
+  });
+  
+  describe('Production', () => {
+    /** @type {import('./test-utils.js').App} */
+    let app;
+    
+    // Mock fetch calls for tracking
+    let fetchCalls = [];
+    const customFetch = async (url, options) => {
```

Comment:
This is incorrect, the custom fetch is supposed to receive a string, not a string and an arbitrary object.